op-service/sources/l1_beacon_client.go

```diff
@@ -102,28 +107,44 @@ func (cl *L1BeaconClient) GetBlobSidecars(ctx context.Context, ref eth.L1BlockRe
 	}
 	slotFn, err := cl.GetTimeToSlotFn(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("%w: failed to get time to slot function", err)
+		return nil, fmt.Errorf("failed to get time to slot function: %w", err)
 	}
 	slot, err := slotFn(ref.Time)
 	if err != nil {
-		return nil, fmt.Errorf("%w: error in converting ref.Time to slot", err)
+		return nil, fmt.Errorf("error in converting ref.Time to slot: %w", err)
 	}
 
 	reqPath := path.Join(sidecarsMethodPrefix, strconv.FormatUint(slot, 10))
-	reqQuery := url.Values{}
-	for i := range hashes {
-		reqQuery.Add("indices", strconv.FormatUint(hashes[i].Index, 10))
+	var reqQuery url.Values
+	if !cl.cfg.FetchAllSidecars {
+		reqQuery = url.Values{}
+		for i := range hashes {
+			reqQuery.Add("indices", strconv.FormatUint(hashes[i].Index, 10))
+		}
 	}
+
 	var resp eth.APIGetBlobSidecarsResponse
 	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
-		return nil, fmt.Errorf("%w: failed to fetch blob sidecars for slot %v block %v", err, slot, ref)
+		return nil, fmt.Errorf("failed to fetch blob sidecars for slot %v block %v: %w", slot, ref, err)
 	}
-	if len(hashes) != len(resp.Data) {
+
+	apiscs := make([]*eth.APIBlobSidecar, 0, len(hashes))
+	// filter and order by hashes
+	for _, h := range hashes {
```

Comment:
You can drop this, the order is not supposed to be guaranteed by this function (see godoc).   Order is enforced later within blobsFromSidecars.

Reply:
Yep I need it for the filtering, so I thought I might as well do it in a order-preserving way. But good callout, will fix godocs and `blobsFromSidecar`

Reply:
done

Reply:
ah damn it, I fixed it and then force pulled or something 🙄 should be fixed now